src/history/HistoryArchive.cpp

```diff
@@ -288,22 +288,93 @@ HistoryArchiveState::containsValidBuckets(Application& app) const
     // Level 0 future buckets are always clear
     assert(currentBuckets[0].next.isClear());
 
-    for (uint32_t i = 1; i < BucketList::kNumLevels; i++)
+    uint32_t minBucketVersion = 0;
+    uint32_t oldestBucketVersion = 0;
+    bool nonEmptySeen = false;
+    Hash const emptyHash;
+
+    auto validateBucketVersion = [&](uint32_t bucketVersion) {
+        if (bucketVersion < minBucketVersion)
+        {
+            CLOG(ERROR, "History")
+                << fmt::format("Incompatible bucket versions: expected version "
+                               "{} or higher, got {}",
+                               minBucketVersion, bucketVersion);
+            return false;
+        }
+        return true;
+    };
+
+    // Process bucket, return version
+    auto getVersion = [&](std::string const& bucketHash) {
+        auto bucket =
+            app.getBucketManager().getBucketByHash(hexToBin256(bucketHash));
+        assert(bucket);
```

Comment:
maybe prefer one of the releaseAssert* macros? Impervious to NDEBUG and now give you backtraces (on some platforms)